VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,537 @@
+
+- ``maven``: Apache Maven http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html
+
+- ``nuget``: NuGet https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges
+  Note that Apache Maven and NuGet are following a similar approach with a
+  math-derived intervals syntax as in https://en.wikipedia.org/wiki/Interval_(mathematics)
+
+- ``gentoo``: Gentoo https://wiki.gentoo.org/wiki/Version_specifier
+
+- ``alpine``: Alpine linux https://gitlab.alpinelinux.org/alpine/apk-tools/-/blob/master/src/version.c
+  (which might be using Gentoo conventions)
+
+- ``generic``: a generic version comparison algorithm (which is TBD, likely a
+  split on punctuation and dealing with digit vs. strings comparisons, like is
+  done in libversion)
+
+
+
+Implementations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Python: https://github.com/nexB/univers
+- Yours!
+
+
+Why not reuse existing version range notations?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Most existing version range notations are tied to a specific version string
+syntax and are therefore not readily applicable to other contexts. For example,
+the use of elements such as tilde and caret ranges in Ruby or Dart notations
+implies that a certain structure exists in the version string (semver or semver-
+like). The inclusion of these additional comparators is a result of the history
+and evolution in a given package ecosystem to address specific needs.
+
+In practice, the unified and reduced set of comparators and syntax defined for
+``vers`` has been designed such that all these notations can be converted to a
+``vers`` and back from a ``vers`` to the original notation. 
+
+In contrast, this would not be possible with existing notations. For instance,
+the Python notation may not work with npm versions and reciprocally.
+
+There are likely to be a few rare cases where round tripping from and to
+``vers`` may not be possible, and in anycase round tripping to and from ``vers``
+may only produce equivalent results and not strictly the same original strings.
+
+Another issue with existing version range notations is that, they are primarily
+meant to be used for dependency constraints and may not readily be reused for
+vulnerable ranges. In particular, a vulnerability may exist for multiple
+"branches" of a given package (such as Django 2, 3 and 4) which may not be
+possible to express with a notation designed exclusively for dependent versions
+resolution.
+
+
+Why not use the NVD CPE Ranges?
+###############################
+
+https://nvd.nist.gov/vuln/vulnerability-detail-pages#divRange
+
+The NVD CPE Match String Range is a complex specification that goes well beyond
+version ranges and is used to match comprehensive configurations across multiple
+products and version ranges. The notation is to uses these fields in the CVE API
+or feeds under "configurations"::
+
+    "versionStartIncluding": "7.3.0",
+    "versionEndExcluding": "7.3.31",
```

Comment:
While looking into https://nvd.nist.gov/vuln/data-feeds#cpeMatch there are also:
```
    "versionStartExcluding" : "9.0.0",
    "versionEndIncluding" : "9.0.46",
```
and concrete enumeration of versions updated daily.